lib/generators/react_on_rails/setup_files_generator.rb

```diff
@@ -0,0 +1,38 @@
+require "rails/generators"
+
+module ReactOnRails
+  module Generators
+    class SetupFilesGenerator < Rails::Generators::Base
+      source_root File.expand_path("../templates", __FILE__)
+
+      def create_assets_symlinks
+        create_link "client/assets/fonts", "../../app/assets/fonts" if Dir.exist?("app/assets/fonts")
+        create_link "client/assets/images", "../../app/assets/images" if
+        Dir.exist?("app/assets/images")
+      end
+
+      def create_jsx_files
+        copy_file "client/app/startup/clientGlobals.jsx",
+                  "client/app/startup/clientGlobals.jsx"
+        copy_file "client/app/startup/serverGlobals.jsx",
+                  "client/app/startup/serverGlobals.jsx"
+      end
+
+      def update_git_ignore
+        append_to_file ".gitignore" do
+          "# React on Rails\nnpm-debug.log\nnode_modules\n\n# Generated js bundles\n/app/assets/javascripts/generated/*"
+        end
+      end
+
+      def update_application_js
+        append_to_file "app/assets/javascripts/application.js" do
```

Comment:
I have one app that uses application.js.coffee and the generator breaks here. Also, one suggestion is to use prepend instead of append, because probably those requires should be on top.

bellow an attempt to add support for application.js.coffee and to show a message if no application.js/js.coffee file is found.

```ruby
      def update_application_js
        data = "\n// It is important that generated/vendor-bundle must be before bootstrap since it is exposing jQuery and jQuery-ujs\n//= require generated/vendor-bundle\n//= require generated/app-bundle\n//= require react_on_rails\n\n"
        if File.exists?("app/assets/javascripts/application.js")
          prepend_to_file "app/assets/javascripts/application.js", data
        elsif File.exists?("app/assets/javascripts/application.js.coffee")
          prepend_to_file "app/assets/javascripts/application.js.coffee", data
        else
          puts "** app/assets/javascripts/application.js was not found. please add the following content to your main javascript file:\n\n#{data}\n\n"
        end
      end


```